server/game/gameSystems/LookMoverDeckCardsTopOrBottomSystem.ts

```diff
@@ -23,59 +20,67 @@ export class LookMoveDeckCardsTopOrBottomSystem<TContext extends AbilityContext
-                            'Put ' + card.internalName + ' on top',
-                            'Put ' + card.internalName + ' to bottom',
-                        ]).flat(),
-                        handlers: cards.map((card: Card) => [(() => {
-                            events.push(moveCard({
-                                target: card,
-                                bottom: false,
-                                destination: Location.Deck
-                            }).generateEvent(context.target, context));
-                            // get rid of the card from cards
-                            cards = cards.filter((a) => a !== card);
-                            choiceHandler(context.player);
-                        }),
-                        (() => {
-                            events.push(moveCard({
-                                target: card,
-                                bottom: true,
-                                destination: Location.Deck
-                            }).generateEvent(context.target, context));
-                            // get rid of the card from cards
-                            cards = cards.filter((a) => a !== card);
-                            choiceHandler(context.player);
-                        })]).flat()
-                    });
-                } else {
+            // Each card has two options to be put on top or on bottom for each option we have a handler whcih
+            // recursively calls the function and removes handlers from the list until the card pool reaches 0.
+            const choiceHandler = (player, cards: any[]) => {
+                if (cards.length === 0) {
                     return true;
                 }
-            };
+                // setup the choices for each card top and bottom
+                const handlerChoices = cards.map((card: Card) => [
+                    'Put ' + card.title + ' on top',
+                    'Put ' + card.title + ' on bottom',
+                ]).flat();
 
-            choiceHandler(context.player);
+                context.game.promptWithHandlerMenu(player, {
+                    activePromptTitle: 'Select card to move to the top or bottom of the deck',
+                    choices: handlerChoices,
+                    handlers: cards.map((card: Card) => [(() => {
+                        this.pushMoveEvent(false, card, cards, events, context, choiceHandler);
+                    }),
+                    (() => {
+                        this.pushMoveEvent(true, card, cards, events, context, choiceHandler);
+                    })]).flat()
+                });
+            };
+            choiceHandler(context.player, cards);
         }
     }
 
+    // Helper method for pushing the move card event into the events array.
+    private pushMoveEvent(bottom: boolean, card: Card, cards: any[], events: GameEvent[], context: TContext,
+        choiceHandler: (player: any, cards: any[]) => boolean) {
```

Comment:
nit: when the param lists get long we typically just do one per line. So like:

```typescript
    private pushMoveEvent(
        bottom: boolean, 
        card: Card, 
        cards: any[], 
        events: GameEvent[], 
        context: TContext,
        choiceHandler: (player: any, cards: any[]) => boolean
    ) {
```

Reply:
made it like that!